### PR TITLE
Admin kubeconfigs sign with client CA and verify with cluster CA

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind bug regression

**What this PR does / why we need it**:
Earlier, kubeconfigs issued via the `shoots/adminkubeconfig` subresource were using the client CA for verifying the `kube-apiserver` server certificate. This is obviously wrong. The client CA should only be used to sign the client certificate, and all known cluster CAs should be put into a bundle and be used to verify the `kube-apiserver`'s server certificate.

**Special notes for your reviewer**:
/milestone v1.45
/cc @acumino @petersutter 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by acumino** [comment]
I tried this locally It works fine as expected.
/lgtm

File: pkg/registry/core/shoot/storage/admin_kubeconfig.go [hunk 1]
```diff
@@ -267,3 +235,79 @@ func findNewestCACertificate(results []*gardencorev1alpha1.GardenerResourceData)
...
+			return nil, fmt.Errorf("could not fetch raw CA data for %q", data.Name)
+		}
+		caBundle = append(caBundle, cert...)
+	}
+
+	if len(caBundle) == 0 {
+		return nil, fmt.Errorf("cluster certificate authority not yet provisioned")
+	}
+
+	return caBundle, nil
```
Comment (petersutter) [inline pkg/registry/core/shoot/storage/admin_kubeconfig.go:265]: in the end `caBundle` should equal to `ca.crt` of secret `<shootname>.ca-cluster`, right?

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by acumino — COMMENTED** [review]
/approve

File: pkg/registry/core/shoot/storage/admin_kubeconfig.go (hunk 1 above)
Comment (rfranzke) [reply pkg/registry/core/shoot/storage/admin_kubeconfig.go:265]: Yes, right 👍🏻

**Pull request merged**